hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestFileUtil.java

```diff
@@ -706,35 +706,40 @@ public void testCreateLocalTempFile() throws IOException {
   public void testUnZip() throws IOException {
     // make sa simple zip
     final File simpleZip = new File(del, FILE);
-    OutputStream os = new FileOutputStream(simpleZip); 
-    ZipOutputStream tos = new ZipOutputStream(os);
-    try {
-      ZipEntry ze = new ZipEntry("foo");
-      byte[] data = "some-content".getBytes("UTF-8");
-      ze.setSize(data.length);
-      tos.putNextEntry(ze);
-      tos.write(data);
-      tos.closeEntry();
-      tos.flush();
-      tos.finish();
-    } finally {
-      tos.close();
-    }
-    
-    // successfully unzip it into an existing dir:
-    FileUtil.unZip(simpleZip, tmp);
-    // check result:
-    assertTrue(new File(tmp, "foo").exists());
-    assertEquals(12, new File(tmp, "foo").length());
-    
-    final File regularFile = new File(tmp, "QuickBrownFoxJumpsOverTheLazyDog");
-    regularFile.createNewFile();
-    assertTrue(regularFile.exists());
-    try {
-      FileUtil.unZip(simpleZip, regularFile);
-      assertTrue("An IOException expected.", false);
-    } catch (IOException ioe) {
-      // okay
+    try (OutputStream os = new FileOutputStream(simpleZip);
+         ZipArchiveOutputStream tos = new ZipArchiveOutputStream(os)) {
+      try {
+        ZipArchiveEntry ze = new  ZipArchiveEntry("foo");
+        ze.setUnixMode(0555);
+        byte[] data = "some-content".getBytes("UTF-8");
+        ze.setSize(data.length);
+        tos.putArchiveEntry(ze);
+        tos.write(data);
+        tos.closeArchiveEntry();
+        tos.flush();
+        tos.finish();
+      } finally {
+        tos.close();
+      }
+
+      // successfully unzip it into an existing dir:
+      FileUtil.unZip(simpleZip, tmp);
+      // check result:
+      assertTrue(new File(tmp, "foo").exists());
+      assertEquals(12, new File(tmp, "foo").length());
+      assertTrue("file lacks execute permissions", new File(tmp, "foo").canExecute());
+      assertFalse("file has write permissions", new File(tmp, "foo").canWrite());
+      assertTrue("file lacks read permissions", new File(tmp, "foo").canRead());
+
+      final File regularFile = new File(tmp, "QuickBrownFoxJumpsOverTheLazyDog");
+      regularFile.createNewFile();
+      assertTrue(regularFile.exists());
+      try {
```

Comment:
use LambdaTestUtils.intercept() here. e.g.

```java
LambdaTestUtils.intercept(IOException.class, () ->
  FileUtil.unZip(simpleZip, regularFile));
```

Reply:
The code for the latest branch has been merged.